contrib/eraser/training/train_eraser.py

```diff
@@ -44,7 +54,245 @@
 from lbm.models.vae import AutoencoderKLDiffusers, AutoencoderKLDiffusersConfig
 from lbm.trainer import TrainingConfig, TrainingPipeline
 from lbm.trainer.utils import StateDictAdapter
+from dataclasses import field
+from neptune.types import File
+
+class EraserLogger(Callback):
+    """
+    Eraser Logger Callback for Neptune.
+
+    Log LPIPS, DISTS, PSNR.
+
+    Use lightning self.log function to log metrics, so 
+    accumulation/distributed logging is handled by lightning.
+
+    Args:
+        num_steps (list[int]): List of number of steps to log metrics for.
+        device (torch.device): Device to use for metrics computation.
+    """
+    def __init__(
+        self, 
+        num_steps: list[int], 
+        device: torch.device | None = None, 
+    ):
+        super().__init__()
+        self.num_steps = num_steps
+        self.device = device  # can be None; we’ll set it dynamically
+        self.metrics = None  # delay initialization
+
+    def setup(self, trainer: Trainer, pl_module: TrainingPipeline, stage=None):
```

Comment:
Since it assumes Neptune only, how about adding an assert like: `assert isinstance(trainer.logger, loggers.NeptuneLogger)`